elfpy/data/contract_interface.py

```diff
@@ -56,15 +69,76 @@ class HyperdriveAddressesJson:
     mock_hyperdrive_math: str = attr.ib()
 
 
-def get_account_balance_for_contract(funding_contract: Contract, account_address: str) -> int:
+def initialize_web3_with_http_provider(ethereum_node: URI | str, request_kwargs: dict | None = None) -> Web3:
+    """Initialize a Web3 instance using an HTTP provider and inject a geth Proof of Authority (poa) middleware.
+
+    Arguments
+    ---------
+    ethereum_node: URI | str
+        Address of the http provider
+    request_kwargs: dict
+        The HTTPProvider uses the python requests library for making requests.
+        If you would like to modify how requests are made,
+        you can use the request_kwargs to do so.
+    """
+    if request_kwargs is None:
+        request_kwargs = {}
+    provider = Web3.HTTPProvider(ethereum_node, request_kwargs)
+    web3 = Web3(provider)
+    web3.middleware_onion.inject(geth_poa.geth_poa_middleware, layer=0)
+    return web3
+
+
+def set_account_balance(web3: Web3, account_address: str, amount_wei: int) -> RPCResponse:
+    """Set an the account using the web3 provider
+
+    Arguments
+    ---------
+    amount_wei : int
+        amount_wei to fund, in wei
+    """
+    # TODO: Assert that we are using anvil (devnet) to be able to call anvil_setBalance
```

Comment:
I think we can remove this.  if we want, we can rename to `set_anvil_account_balance`.  I don't think we'll be using any other dev nodes and this will obviously fail anyway on goerli or mainnet.